internal/usecase/interactor/layer.go

```diff
@@ -365,6 +366,11 @@ func (i *Layer) AddGroup(ctx context.Context, inp interfaces.AddLayerGroupInput,
 		if rf := ds.FieldRef(representativeFieldID); rf != nil && rf.Type() == dataset.ValueTypeString {
 			name = rf.Value().Value().(string)
 		}
+		if name == "" && layerTitleFieldId != nil {
+			if rf := ds.FieldRef(layerTitleFieldId); rf != nil && rf.Type() == dataset.ValueTypeString {
+				name = rf.Value().Value().(string)
+			}
+		}
```

Comment:
```suggestion
```

representativeFieldID is already available for such usecase, so just set layerTitleFieldId as representativeFieldID instead of setting layer name manually.

Pass the representativeFieldID to the logic that generates the dataset schema and set the representativeFieldID internally.

Reply:
We can not set the representativeFieldID in the dataset schema, because according to the feature design the user should be able to set the representativeFieldID every time he adds the dataset as layer group

Reply:
Oh, I see. I was mistaken.